### DISCUSSION TITLE
calcOrientations function

### DISCUSSION THREAD

**Discussion by lampardlamps** [mtex-toolbox/mtex#1408] [Feature Requests]
First of all - you've probably heard this a million times - thank you so much for creating such an amazing tool for the community!

I would like to ask about the `calcOrientations` function. As I understand it, this is used to randomly sample the ODF in the orientation space to create a list of individual orientations. This is useful for analysing, say, data from X-ray or neutron diffraction tests, which do not have spatial orientation information, and creating actual orientations for simulation models etc.
However, random sampling can also lead to inaccurate representations. For any strong texture, the preferred orientations should have more orienation data points, and random sampling would diminish that texture (some times significantly). I'm attaching a figure where on the left is the pole figure drawn from 10M data points from an EBSD test, in the middle is from a orientation data set of 160k points, random sampled from the ODF using `calcOrientations`, and on the right is from a data set of 160k randomly sampled from the original EBSD results. You can see the diminishing texture in the middle plot clearly. 

![MicrosoftTeams-image (7)](https://user-images.githubusercontent.com/1702296/164506438-f81a71f5-f072-4d74-ac0c-f3836ebe6a40.png)

So I'm wondering whether there is a better way of sampling the ODF to create more representative orientations. The only thing I can think of is to do weighted sampling in the orientation space, and sample more data points where the orientation is preferred. But it seems to be a tricky thing to implement, if possible at all. Any tips and advice is most appreciated, many thanks indeed.

**Comment by kilir** [comment]
Hi,
that is what `discreteSample` (which is what calcOrientations executes) actually does: sampling points according to the their probability. I think your observation rather stems from the density estimator you used.
If you have the "true" distribution and you sample 100 or 1e6 points, obviously the 10 points will more poorly reproduce the positions/shape of your odf, hence  - and in order not to locally overestimate your texture - you'd choose a larger kernel. This will obviously result in a lower texture strength. If you have more points, you have more information about the "true" function allowing your to do a more reliable estimate of the true function.

If you try the kernel estimators (`calcKernel()`), most methods are rather conservative in the sense that they should not tend to overestimate densities and should not produce local maxima which may not belong to the "true" function. The default 'KLCV' is probably the least timid one. and often a good shot.

Hope this helps.
Cheers,
Rüdiger

**Reply by lampardlamps** [reply]
Dear Rüdiger,
  Thank you for your reply. I didn't realise that the sampling was already done on a probabilistic basis, sorry. In my case, I was getting 160k data points, which I think is plenty to display a statistical fidelity? Sorry I was not sure I understand your comments on `calcKernel`, what would be your suggestion for me to increase the respresentative accuracy please? Cheers.
